src/framework/auth/services/token.service.ts

```diff
@@ -105,9 +129,17 @@ export class NbTokenService {
         return observableOf(this.tokenWrapper);
       },
 
+      // TODO is it possible to unify interface
       setter: (token: string | NbAuthSimpleToken): Observable<null> => {
-        const raw = token instanceof NbAuthSimpleToken ? token.getValue() : token;
-        localStorage.setItem(this.getConfigValue('token.key'), raw);
+        let tokenValue;
```

Comment:
Let's get gid of one `else` clause for simplicity 
```
if (!(token instanceof NbAuthSimpleToken)) {
  this.tokenWrapper.setValue(token); // ideally in future, we do this `token = this.tokenWrapper.createToken(token);`
  token = this.tokenWrapper;
}
localStorage.setItem(this.getConfigValue('token.key'), token.toString());
```

Reply:
fixed